vNext/src/Microsoft.AspNet.OData/Query/ODataQueryOptions.cs

```diff
@@ -97,6 +115,25 @@ public virtual IQueryable ApplyTo(IQueryable query, ODataQuerySettings querySett
                 query = SelectExpand.ApplyTo(query, querySettings, _assemblyProvider);
             }
 
+            if (Skip.HasValue)
+            {
+                query = ExpressionHelpers.Skip(query, Skip.Value, Context.ElementClrType, false);
+            }
+
+            int? take = null;
+            if (querySettings.PageSize.HasValue)
+            {
+                take = Math.Min(querySettings.PageSize.Value, int.MaxValue);
+            }
+            if (Top.HasValue)
+            {
+                take = Math.Min(Top.Value, take ?? int.MaxValue);
+            }
+            if (take.HasValue)
+            {
+                query = ExpressionHelpers.Take(query, take.Value, Context.ElementClrType, false);
```

Comment:
The ExpressionHelpers class uses reflection to invoke the LINQ extension methods Skip and Take respectively, and those methods both accept int arguments, not longs.  If we want to promote our $skip and $top parameters to support long integers, we'd need to implement our own edition of the Skip and Take methods.  Thoughts?